commcare_connect/form_receiver/tests/test_receiver_integration.py

```diff
@@ -596,6 +587,44 @@ def test_reciever_verification_flags_catchment_areas(
     assert ["catchment", "Visit outside worker catchment areas"] in visit.flag_reason.get("flags", [])
 
 
+@pytest.mark.parametrize("opportunity", [{"managed": True}], indirect=True)
+def test_receiver_auto_agree_approved_visit(
+    mobile_user_with_connect_link: User, api_client: APIClient, opportunity: Opportunity
+):
+    assert opportunity.managed
+    form_json = get_form_json_for_payment_unit(opportunity.paymentunit_set.all()[0])
+    make_request(api_client, form_json, mobile_user_with_connect_link)
+    visit = UserVisit.objects.get(user=mobile_user_with_connect_link)
+    assert not visit.flagged
+    assert visit.status == VisitValidationStatus.approved
+    assert visit.review_status == VisitReviewStatus.agree
+
+
+@pytest.mark.parametrize("opportunity", [{"managed": True}], indirect=True)
+def test_receiver_flagged_visit_review_pending(
+    mobile_user_with_connect_link: User, api_client: APIClient, opportunity: Opportunity
+):
+    assert opportunity.managed
+    form_json = get_form_json_for_payment_unit(opportunity.paymentunit_set.all()[0])
+    form_json["metadata"]["location"] = None
+    make_request(api_client, form_json, mobile_user_with_connect_link)
+    visit = UserVisit.objects.get(user=mobile_user_with_connect_link)
+    assert visit.flagged
+    assert visit.status == VisitValidationStatus.pending
+    assert visit.review_status == VisitReviewStatus.pending
+
+
+def get_form_json_for_payment_unit(payment_unit):
```

Comment:
We have too many get-form-json methods, wonder if they can be combined.

Reply:
Yeah, I was thinking the same. I am in favor of keeping the latest methods that were created including the fixture that you created for mobile user. That will simplify all these tests a lot.